cogs/actions_tracking.py

```diff
@@ -0,0 +1,270 @@
+        required=True,
+        parameter_name="str_action_member_id",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def action(self, ctx: TeXBotApplicationContext, str_action_member_id: str, str_action_description: str) -> None:  # noqa: E501
+        """
+        Definition and callback response of the "action" command.
+
+        The action command adds an action to the specified user. If no user is specified
+        we assume the action is aimed at the command issuer.
+        """
+        try:
+            action: Action = await Action.objects.acreate(
+                member_id=int(str_action_member_id),
+                description=str_action_description,
+            )
+        except ValidationError as create_action_error:
+            error_is_already_exits: bool = (
+                "__all__" in create_action_error.message_dict
+                and any (
+                    "already exists" in error
+                    for error
+                    in create_action_error.message_dict["__all__"]
+                )
+            )
+            if not error_is_already_exits:
+                await self.command_send_error(ctx, message="An unrecoverable error occured.")
+                logger.critical(
+                    "Error upon creating Action object: %s",
+                    create_action_error,
+                )
+                await self.bot.close()
+                return
+
+            await self.command_send_error(
+                ctx,
+                message="You already have an action with that description!",
+            )
+            return
+
+        await ctx.respond(f"Action: {action.description} created for user: <@{str_action_member_id}>")  # noqa: E501
+
+    @discord.slash_command( # type: ignore[no-untyped-call, misc]
+        name="list-user-actions",
+        description="Lists all actions for a specified user",
+    )
+    @discord.option( # type: ignore[no-untyped-call, misc]
+        name="user",
+        description="The user to list actions for.",
+        input_type=str,
+        autocomplete=discord.utils.basic_autocomplete(action_autocomplete_get_committee), # type: ignore[arg-type]
+        required=True,
+        parameter_name="str_action_member_id",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def list_user_actions(self, ctx:TeXBotApplicationContext, str_action_member_id: str) -> None:  # noqa: E501
+        """
+        Definition and callback of the list user actions command.
+
+        Takes in a user and lists out their current actions.
+        """
+        actions = [action async for action in Action.objects.all() if Action.hash_member_id(str_action_member_id) == action.hashed_member_id]  # noqa: E501
```

Comment:
Does this hash need to be calculated each time? Also is there a better way to find entities than fetching them all and iterating over them?